src/main/java/meli/freshfood/service/BatchServiceImpl.java

```diff
@@ -102,12 +99,43 @@ public void updateStock(ProductDTO productDTO) {
         });
     }
 
+    @Override
+    public List<BatchStockDTO> filterNotExpiredProductsByDays(List<BatchStockDTO> batches, Integer intervalDate) {
+        return batches.stream().filter((b) -> {
+            LocalDate dueDate = b.getDueDate();
+            if (dueDate.isAfter(LocalDate.now()) &&
+                    dueDate.isBefore(LocalDate.now().plusDays(intervalDate))) {
+                return true;
+            } else {
+                return false;
+            }
+        }).collect(Collectors.toList());
+    }
+
+    //Retorna todos os lotes armazenados em um setor de um armazém dentro de um intervalo de datas filtrados pela data de vencimento
+    @Override
+    public List<BatchStockDTO> findBatchesFilteredByDueDateAndSection(Integer intervalDate, Long id) {
+        List<Batch> batches = batchRepository.findAll();
+        List<Batch> batchesSorted;
+
+        if (id != null) {
```

Comment:
acredito que pode ser alterado para tratar apenas quando o id for diferente de null e fazer o sort antes de entrar nele, já que o filtro não irá alterar a ordem.

Ficaria algo nesse sentido:

batchesSorted = sortByDueDate(batches);
if(id != null) {
   batchesSorted = filterBatchesBySection(batchesSorted, id);
}